plugins/cloudforecast/src/hooks.tsx

```diff
@@ -0,0 +1,114 @@
+import { useEffect, useState } from "react";
+import { usePluginContext } from "@cortexapps/plugin-core/components";
+import {
+  type CloudForecastData,
+  isCloudForecastData,
+} from "./cloudForecastSchema";
+
+export interface UseCloudForecastPluginConfig {
+  isLoading: boolean;
+  cloudForecastDataKey: string;
+}
+
+export const useCloudForecastPluginConfig =
+  (): UseCloudForecastPluginConfig => {
+    const { apiBaseUrl } = usePluginContext();
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [cloudForecastDataKey, setCloudForecastDataKey] =
+      useState<string>("");
+
+    useEffect(() => {
+      if (!apiBaseUrl) {
+        return;
+      }
+
+      const fetchPluginConfig = async (): Promise<void> => {
+        setIsLoading(true);
+        let newCloudForecastDataKey = "cloudforecast";
+        try {
+          const response = await fetch(
+            `${apiBaseUrl}/catalog/cloudforecast-plugin-config/openapi`
+          );
+          const data = await response.json();
+          const cloudForecastDataKeyFromEntity =
+            data.info["x-cortex-definition"]["cloudforecast-data-key"];
+          if (cloudForecastDataKeyFromEntity) {
+            newCloudForecastDataKey = cloudForecastDataKeyFromEntity;
+          }
+        } catch (e) {
```

Comment:
I think it would be helpful to at least output the error message to console.error() so that there's some feedback as to why the API call failed

Reply:
so specifically here I don't want to set an error state if the plugin config does not exist because I have a default and the plugin config entity only needs to exist if you are overriding the default.... compare to `useEntityCloudForecastData` where i do want to set an error state to show to the user if the data is invalid or not found

Reply:
Ah makes sense, thanks!